tests/test_core.py

```diff
@@ -23,15 +24,112 @@ class test_PairMatrix(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different list types."""
-        raise NotImplementedError()
+
+        pair_types = ('A','B')
+        pair_dict = {('A','B'):{}, ('B','B'):{}, ('A','A'):{}}
+
+        #test construction with tuple input
+        m = relentless.core.PairMatrix(types=('A','B'))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        #test construction with list input
+        m = relentless.core.PairMatrix(types=['A','B'])
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        #test construction with numpy array input
+        m = relentless.core.PairMatrix(types=np.array(['A','B']))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        pair_types = ('A',)
+        pair_dict = {('A','A'):{}}
+
+        #test construction with single type tuple
+        m = relentless.core.PairMatrix(types=('A',))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
 
     def test_accessors(self):
         """Test get and set methods on pairs."""
-        raise NotImplementedError()
+
+        m = relentless.core.PairMatrix(types=('A','B'))
+
+        #test set and get for each pair type
+        m['A','A']['energy'] = 1.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','B'], {})
+        self.assertEqual(m['B','B'], {})
+
+        m['A','B']['energy'] = -1.0
```

Comment:
Can you do a test with key `m['B','A']`, which should give the exact same value as `m['A','B']`?